File: python/questao15.py
def eh_palindromo(numero):
    numero_str = str(numero)
    invertido_str = numero_str[::-1]
    return numero_str == invertido_str

def main():
    numero = int(input("Digite um número: "))
    
    if eh_palindromo(numero):
        print(f"{numero} é um palíndromo.")
    else:
        print(f"{numero} não é um palíndromo.")

if __name__ == "__main__":
    main()

File: c/questao15.c
#include <stdio.h>

int eh_palindromo(int numero) {
    int original = numero;
    int invertido = 0;

    while (numero > 0) {
        int digito = numero % 10;
        invertido = invertido * 10 + digito;
        numero = numero / 10;
    }

    return original == invertido;
}

int main() {
    int numero;

    printf("Digite um número: ");
    scanf("%d", &numero);

    if (eh_palindromo(numero)) {
        printf("%d é um palíndromo.\n", numero);
    } else {
        printf("%d não é um palíndromo.\n", numero);
    }

    return 0;
}
